contracts/test/veLON.t.sol

```diff
@@ -0,0 +1,265 @@
+        stakerLon.assertChange(-int256(stakeAmount));
+        lockedLon.assertChange(int256(stakeAmount));
+        // assertEq(veLon.totalNFTSupply(), numberOfNftBefore + 1);
+        // // this part is weird
+        // uint256 increasedPower = _vePowerAdd(stakeAmount, lockDuration);
+        // assertEq(veLon.totalSupply(), totalVePowerBefore + increasedPower);
+        // assertEq(veLon.balanceOfNFT(tokenId), increasedPower);
+        vm.stopPrank();
+
+        return tokenId;
+    }
+
+    /*********************************
+     *         Test: deposit         *
+     *********************************/
+    function testCreateLock() public {
+        _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, MAX_LOCK_TIME);
+    }
+
+    function testFuzzCreateLock(uint256 _amount) public {
+        vm.prank(user);
+        vm.assume(_amount <= 100 * 1e18);
+        vm.assume(_amount > 0);
+        veLon.createLock(_amount, MAX_LOCK_TIME);
+    }
+
+    function testCannotCreateLockWithZero() public {
+        vm.prank(user);
+        vm.expectRevert("Zero lock amount");
+        veLon.createLock(0, MAX_LOCK_TIME);
+    }
+
+    function testDepositFor() public {
+        uint256 tokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, MAX_LOCK_TIME);
+        require(tokenId != 0, "No lock created yet");
+
+        stakerLon = BalanceSnapshot.take(user, address(lon));
+        lockedLon = BalanceSnapshot.take(address(veLon), address(lon));
+        vm.prank(user);
+        veLon.depositFor(tokenId, 1);
+
+        //TODO check if locked.amount and locked.end have increased
+        stakerLon.assertChange(-(1));
+        lockedLon.assertChange(1);
+
+        vm.stopPrank();
+    }
+
+    function testDepositForByOther() public {
+        uint256 tokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, MAX_LOCK_TIME);
+        vm.prank(other);
+        veLon.depositFor(tokenId, 100);
+    }
+
+    function testFuzzDepositFor(uint256 _amount) public {
+        uint256 tokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, MAX_LOCK_TIME);
+        vm.prank(user);
+        vm.assume(_amount <= 50 * 1e18);
+        vm.assume(_amount > 0);
+        veLon.depositFor(tokenId, _amount);
+    }
+
+    function testExtendLock() public {
+        uint256 tokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, 1 weeks);
+        uint256 lockEndBefore = veLon.unlockTime(tokenId);
+
+        vm.prank(user);
+        veLon.extendLock(tokenId, 2 weeks);
+        uint256 lockEndAfter = veLon.unlockTime(tokenId);
+
+        require((lockEndAfter - lockEndBefore) == 1 weeks, "wrong time extended");
+    }
+
+    function testCannoExtendLockExpired() public {
+        uint256 lockTime = 1 weeks;
+        uint256 tokenId = _stakeAndValidate(user, DEFAULT_STAKE_AMOUNT, lockTime);
+
+        //pretend 1 week has passed and the lock expired
+        vm.warp(block.timestamp + 1 weeks);
+        vm.prank(user);
+        vm.expectRevert("Lock expired");
```

Comment:
We also need to test `"Not approved or owner" revert` for `extendLock` function.